STACpopulator/stac_utils.py

```diff
@@ -62,9 +70,100 @@ def collection2literal(collection):
     return typing.Literal[terms]
 
 
-class STACItem(BaseModel):
-    start_datetime: dt.datetime
-    end_datetime: dt.datetime
+class AutoValueEnum(Enum):
+    def _generate_next_value_(  # type: ignore
+        name: str, start: int, count: int, last_values: List[Any]
+    ) -> Any:
+        return name
+
+
+# DH: There is a question here whether we want to use pystac.Item or not.
+# pystac.Item takes datetime, start_datetime and end_datetime as optional parameters, and then copies them into
+# properties.
+# If we use pystac.Item, we don't have to put start_datetime and end_datetime into Properties, we can let pystac do
+# that.
+class ItemProperties(BaseModel):
+    start_datetime: Optional[dt.datetime] = None
+    end_datetime: Optional[dt.datetime] = None
+    datetime: Optional[dt.datetime] = None
+
+    @field_validator("datetime", mode="before")
+    def validate_datetime(cls, v: Union[dt.datetime, str], values: Dict[str, Any]) -> dt:
+        if v == "null":
+            if not values["start_datetime"] and not values["end_datetime"]:
+                raise ValueError(
+                    "start_datetime and end_datetime must be specified when datetime is null"
+                )
+
+
+class Geometry(TypedDict):
+    type: str
+    coordinates: List[List[List[float]]]
+
+class Asset(BaseModel):
+    href: AnyHttpUrl
+    media_type: Optional[str] = None
+    title: Optional[str] = None
+    description: Optional[str] = None
+    roles: Optional[List[str]] = None
+
+class Link(BaseModel):
+    """
+    https://github.com/radiantearth/stac-spec/blob/v1.0.0/collection-spec/collection-spec.md#link-object
+    """
+
+    href: str = Field(..., alias="href", min_length=1)
+    rel: str = Field(..., alias="rel", min_length=1)
+    type: Optional[str] = None
+    title: Optional[str] = None
+    # Label extension
+    label: Optional[str] = Field(None, alias="label:assets")
+    model_config = ConfigDict(use_enum_values=True)
+
+    def resolve(self, base_url: str) -> None:
+        """resolve a link to the given base URL"""
+        self.href = urljoin(base_url, self.href)
+
+class PaginationMethods(str, AutoValueEnum):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    GET = auto()
+    POST = auto()
+
+
+class PaginationRelations(str, AutoValueEnum):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    next = auto()
+    previous = auto()
+
+class PaginationLink(Link):
+    """
+    https://github.com/radiantearth/stac-api-spec/blob/master/api-spec.md#paging-extension
+    """
+
+    rel: PaginationRelations
+    method: PaginationMethods
+    body: Optional[Dict[Any, Any]] = None
+    merge: bool = False
+
+Links = RootModel[List[Union[PaginationLink, Link]]]
+
+
+class Item(BaseModel):
+    id: str = Field(..., alias="id", min_length=1)
+    geometry: Optional[Geometry] = None
```

Comment:
I don't think `geometry, bbox, properties` (maybe `assets` too?) should be optional. Not sure about `collection` and `datetime`. 

Also, I don't think we need `datetime` for the Item class since `start_datetime` and `end_datetime` are more appropriate for CMIP6 data and those are included in the `ItemProperties` data model.

Reply:
The [specs](https://github.com/radiantearth/stac-spec/blob/master/item-spec/json-schema/item.json) mention that one of geometry or bbox is mandatory.